04.bowling/bowling.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+shots = []
+ARGV[0].split(',').each do |score|
+  if score == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << score.to_i
+  end
+end
+
+frames = []
+shots.each_slice(2) do |score|
+  frames << score
+end
+
+point = 0
+frames.each_with_index do |frame, i|
```

Comment:
sumメソッドを活用すると、こんなふうに書けるかも？

```suggestion
point = frames.each_with_index.sum do |frame, i|
```

Reply:
すみません、具体的な完成イメージが把握できませんでした。

framesが2次元配列で、かつ各frameの合計やインデックスの条件で処理を分けているので、10行目部分でsumをすることは出来ないのでは？と思います。

Reply:
> すみません、具体的な完成イメージが把握できませんでした。

この部分ですが、masanaさんの対応は以下のどれになりますか？

1. イメージがわかないので、コードは何もいじってない
2. `.sum`を付けてみたがエラーが出た（または点数がおかしくなった）。その上でロジックのイメージが沸かなかった
3. その他

もし1であれば、実際にコードを触ってみてほしいです。
その上で、「こんなコードを書いたら、こんな結果になりました（自分の書いたコードと、実行結果を載せる）。○○の部分をどうしたらいいのかわかりません」というように質問してきてください。

ちなみに、僕も手元で動かしてみましたが、`frames.each_with_index.sum do |frame, i|`の形で問題なく点数計算することができました。（ブロック内のロジックには多少手を加える必要はありますが）

なので、実際に動かしてデバッグしていけば、ある程度完成形のイメージはできるんじゃないかな〜と思ってます。

Reply:
説明不足ですみません。
実行するとエラーが出ていたこと、かつ`.sum`を付ける理由の理解が曖昧な状態でした。

当初`.sum`を付けて実行した結果`next unless frame.sum == 10 && i < 9`の部分で、

bowling.rb:9:in `+': nil can't be coerced into Integer (TypeError)

のエラーが発生していました。
調べた結果、nextに引数を指定しないとnilが返ってしまい、繰り返し処理にて加算処理ができないとのことでした。
そのため引数として`next frame.sum` を指定し、正しく処理されるようになりました。